reproject/spherical_intersect/core.py

```diff
@@ -130,69 +134,49 @@ def _reproject_celestial(array, wcs_in, wcs_out, shape_out, parallel=True, _meth
     # raw C function, otherwise Cython might complain.
-    from numpy import ascontiguousarray as aca
-    from ._overlap import _reproject_slice_cython
-    common_func_par = [0, ny_in, nx_out, ny_out, aca(xp_inout), aca(yp_inout), aca(xw_in), aca(yw_in), aca(xw_out), aca(yw_out), aca(array), shape_out]
 
-    # Abstract the serial implementation in a separate function so we can reuse it.
-    def serial_impl():
-        array_new, weights = _reproject_slice_cython(0, nx_in, *common_func_par)
+    aca = np.ascontiguousarray
+    common_func_par = [0, ny_in, nx_out, ny_out, aca(xp_inout), aca(yp_inout),
+                       aca(xw_in), aca(yw_in), aca(xw_out), aca(yw_out), aca(array),
+                       shape_out]
 
-        array_new /= weights
+    if nproc == 1:
+
+        array_new, weights = _reproject_slice([0, nx_in] + common_func_par)
+
+        with np.errstate(invalid='ignore'):
+            array_new /= weights
 
         return array_new, weights
 
-    if _method == "c" and nproc == 1:
-        return serial_impl()
+    elif (nproc is None or nproc > 1):
 
-    # Abstract the parallel implementation as well.
-    def parallel_impl(nproc):
         from multiprocessing import Pool, cpu_count
+
         # If needed, establish the number of processors to use.
         if nproc is None:
             nproc = cpu_count()
 
-        # Create the pool.
-        pool = None
-        try:
-            # Prime each process in the pool with a small function that disables
-            # the ctrl+c signal in the child process.
-            pool = Pool(nproc, _init_worker)
-
-            # Accumulator for the results from the parallel processes.
-            results = []
-
-            for i in range(nproc):
-                start = int(nx_in) // nproc * i
-                end = int(nx_in) if i == nproc - 1 else int(nx_in) // nproc * (i + 1)
-                results.append(pool.apply_async(_reproject_slice_cython, [start, end] + common_func_par))
-
-            array_new = sum([_.get()[0] for _ in results])
-            weights = sum([_.get()[1] for _ in results])
-
-        except KeyboardInterrupt:  # pragma: no cover
-            # If we hit ctrl+c while running things in parallel, we want to terminate
-            # everything and erase the pool before re-raising. Note that since we inited the pool
-            # with the _init_worker function, we disabled catching ctrl+c from the subprocesses. ctrl+c
-            # can be handled only by the main process.
-            if not pool is None:
-                pool.terminate()
-                pool.join()
-                pool = None
-            raise
-
-        finally:
-            if not pool is None:
-                # Clean up the pool, if still alive.
-                pool.close()
-                pool.join()
+        # Prime each process in the pool with a small function that disables
+        # the ctrl+c signal in the child process.
+        pool = Pool(nproc, _init_worker)
```

Comment:
I did put print statements.
The crash happens a few lines below: the `pool.close()` call in line 172.